packages/state-utils/src/create-selector/index.ts

```diff
@@ -16,38 +16,39 @@ import warn from '@wordpress/warning';
 /**
  * Defines acceptable argument types for a memoized selector when using the
  * default cache key generating function.
- *
- * @type {Array}
  */
-const VALID_ARG_TYPES = [ 'number', 'boolean', 'string' ];
+const VALID_ARG_TYPES = [ 'number', 'boolean', 'string' ] as string[];
+
+type Dependant< TState, TProps extends any[], TDependency > = (
+	state: TState,
+	...props: TProps
+) => TDependency;
 
 /**
  * Default behavior for determining whether current state differs from previous
  * state, which is the basis upon which memoize cache is cleared. Should return
  * a value or array of values to be shallowly compared for strict equality.
  *
- * @type    {Function}
- * @param   {object}    state Current state object
- * @returns {(Array|*)}       Value(s) to be shallow compared
+ * @param state Current state object
+ * @returns Value(s) to be shallow compared
  */
-const DEFAULT_GET_DEPENDANTS = ( state ) => state;
+const DEFAULT_GET_DEPENDANTS: Dependant< any, any, any > = < TState >( state: TState ): TState =>
```

Comment:
I wonder if we can make this a bit more specific than `any, any, any`.

Reply:
Unfortunately we cannot because the default literally operates on any state. anything more specific and we wouldn't be able to serve as the default for _any_ selector.